outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -247,6 +262,85 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
         );
     }
 
+    private getNoCompanyInsightsSection = () => {
+        let NoInsightsSection = null;
+        const company = this.state.company as Company;
+        if (company.isEmpty())
```

Comment:
Is there really no templating mechanism?
Doing all of this in the code is very tedious and not readable.

Reply:
Unfortunately there is no templating mechanism in react, react relies on JSX to display html code, this has the advantage of being able to easily mix between HTML and JavaScript code, but it also has some drawbacks.